wordpress-mobile/release-toolkit
# Fix: Typography dashes substitution in Android

The Android linter includes some typography checks and, in particular, it checks for the use of en-dashes in numeric ranges (see `Typography Dashes` [here](http://tools.android.com/tips/lint-checks)).

This rule is not always applied by translators with the result that the build fails because of the related error raised by the linter. 
The old bash script that was used to download the translations had a check for this case and some logic to substitute the dash on the fly during the download, but a similar behaviour hasn't been implemented in the new Ruby script. 

This PR fixes the issue by implementing the substitution logic. The detection code follows the implementation in the Android linter (see `mCheckDashes` https://android.googlesource.com/platform/tools/base/+/master/lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/TypographyDetector.java#262).

## To Test
1. Verify CI is green.
2. The following step can be performed on any client, though I'm using WPAndroid as a reference to make things clearer: 
2.a Checkout a new branch from `develop`.
2.b [Optional] Edit `Fastfile` to add `skip_commit: true` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/a6fad4f9869d221b51ddcc643761b88aaba65458/fastlane/Fastfile#L679) to avoid unintentional commits. 
2.c Run `bundle exec fastlane download_translations`.
2.d Verify that the lane fails because of the `TypographyDashes` error from the linter (it does as of today because some strings on GlotPress contains the wrong dash).
2.e [Optional]You may want to commit the state of the branch now, to make it easier to spot the differences after the fix.
2.f Update `Pluginfile` to use the HEAD commit of this branch. 
2.g Run `bundle exec fastlane download_translations` again.
2.h Verify that the lane succeeds.
2.i Verify that the only diff is related to the en-dash in numeric range.
